src/Symfony/Component/Security/Http/RememberMe/PersistentTokenBasedRememberMeServices.php

```diff
@@ -125,4 +128,18 @@ protected function onLoginSuccess(Request $request, Response $response, TokenInt
             )
         );
     }
+
+    private function generateHash(string $tokenValue): string
+    {
+        return self::HASHED_TOKEN_PREFIX.hash_hmac('sha256', $tokenValue, $this->getSecret());
+    }
+
+    private function isTokenValueValid(string $tokenValue, PersistentTokenInterface $persistentToken): bool
+    {
+        if (self::HASHED_TOKEN_PREFIX === substr($persistentToken->getTokenValue(), 0, \strlen(self::HASHED_TOKEN_PREFIX))) {
```

Comment:
We use strpos() in similar situations. The other option is strncmp(). Using substr() leads to memcopies, which are better avoided when possible.

Reply:
Ok, updated with strpos().